### PR TITLE
Fix flaky test

### PR DESCRIPTION
**Pull request by guperrot**
Please have a look at our [guidelines for contributions](https://github.com/microsoft/appcenter-sdk-android/blob/develop/CONTRIBUTING.md) and consider the following before you submit the PR:

* [ ] Has `CHANGELOG.md` been updated?
* [x] Are tests passing locally?
* [x] Are the files formatted correctly?
* [x] Did you add unit tests?
* [ ] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?
* [ ] Did you check UI tests on the sample app? They are not executed on CI.

## Description

Trying to fix a flaky test.

### CODE REVIEW

**Comment by guperrot** [comment]
reopened because the build never started

**Comment by guperrot** [comment]
Screen Shot 2019-10-17 at 5 53 34 PM

**Comment by MatkovIvan** [comment]
@guperrot Could you please mention somewhere what exactly test is flaky?
Does it fix `saveWrapperException`?

File: sdk/appcenter/src/test/java/com/microsoft/appcenter/http/DefaultHttpClientTest.java [hunk 1]
```diff
@@ -673,16 +675,43 @@ public void cancelCurrentCallsOnClose() throws Exception {
...
         /* Mock AsyncTask. */
         String urlString = "https://mock/get";
-        DefaultHttpClientCallTask mockCall = mock(DefaultHttpClientCallTask.class);
-        whenNew(DefaultHttpClientCallTask.class).withAnyArguments().thenReturn(mockCall);
+        final AtomicReference<DefaultHttpClientCallTask> callTask = new AtomicReference<>();
+        whenNew(DefaultHttpClientCallTask.class).withAnyArguments().thenAnswer(new Answer<Object>() {
+
+            @Override
+            public Object answer(InvocationOnMock invocation) {
+
```
Comment (MatkovIvan) [inline sdk/appcenter/src/test/java/com/microsoft/appcenter/http/DefaultHttpClientTest.java]:
```suggestion
```

**Comment by guperrot** [comment]
> @guperrot Could you please mention somewhere what exactly test is flaky?
> Does it fix `saveWrapperException`?

Yeah at first it was that, then disabling log sending in the android test uncovered some lines in the http logic.

**Pull request merged**